python/semantic_kernel/connectors/data/azure_ai_search/azure_ai_search_vector_record_store.py

```diff
@@ -0,0 +1,180 @@
+            )
+        except ValidationError as exc:
+            raise MemoryConnectorInitializationError("Failed to create Azure Cognitive Search settings.") from exc
+
+        super().__init__(
+            data_model_type=data_model_type,
+            data_model_definition=data_model_definition,
+            collection_name=azure_ai_search_settings.index_name,
+            kernel=kernel,
+        )
+        self._search_index_client = search_index_client or get_search_index_async_client(
+            azure_ai_search_settings=azure_ai_search_settings,
+            azure_credential=azure_credentials,
+            token_credential=token_credentials,
+        )
+        if self.collection_name:
+            self._search_clients: dict[str, SearchClient] = {
+                self.collection_name: self._search_index_client.get_search_client(self.collection_name)
+            }
+        else:
+            self._search_clients = {}
+
+    async def close(self):
+        """Async close connection, invoked by MemoryStoreBase.__aexit__()."""
+        for search_client in self._search_clients.values():
+            await search_client.close()
+        await self._search_index_client.close()
+        self._search_clients.clear()
+
+    def _get_search_client(self, collection_name: str | None = None) -> SearchClient:
+        """Create or get a search client for the specified collection."""
+        collection_name = self._get_collection_name(collection_name)
+        if collection_name not in self._search_clients:
+            self._search_clients[collection_name] = self._search_index_client.get_search_client(collection_name)
+        return self._search_clients[collection_name]
+
+    @override
+    async def upsert(
+        self,
+        record: object,
+        collection_name: str | None = None,
+        generate_embeddings: bool = True,
+        **kwargs: Any,
+    ) -> str | None:
+        result = await self.upsert_batch([record], collection_name, generate_embeddings, **kwargs)
+        return result[0] if result else None
+
+    @override
+    async def upsert_batch(
+        self,
+        records: OneOrMany[TModel],
+        collection_name: str | None = None,
+        generate_embeddings: bool = False,
+        **kwargs,
+    ) -> list[str] | None:
+        if generate_embeddings:
+            await self._add_vector_to_records(records)
+
+        result = await self._get_search_client(collection_name).merge_or_upload_documents(
+            documents=self._convert_model_to_list_of_dicts(records), **kwargs
+        )
+        if result[0].succeeded:
+            return [res.key for res in result]
+        return None
+
+    @override
+    async def get(self, key: str, collection_name: str | None = None, **kwargs) -> TModel:
+        if self._container_mode:
+            return await self.get_batch([key], collection_name, **kwargs)
+        return (await self.get_batch([key], collection_name, **kwargs))[0]
```

Comment:
If both branches are making the same call `await self.get_batch([key], collection_name, **kwargs)`, would it make sense to call it before the condition and then return either the full result or the first element of the result in the condition?
```python
batch = await self.get_batch([key], collection_name, **kwargs)
if self._container_mode:
    return batch
return batch[0]
```